src/back-end/lib/db/counter.ts

```diff
@@ -0,0 +1,66 @@
+import { tryDb } from 'back-end/lib/db';
+import { valid } from 'shared/lib/http';
+
+interface ViewCounter {
+  name: string;
+  count: number;
+}
+
+export const incrementCounters = tryDb<[string[]], Record<string, number>>(async (connection, names) => {
+  // Update existing counters
+  const existingCounters = await connection<ViewCounter>('viewCounters')
+    .whereIn('name', names)
+    .increment('count')
+    .update({}, 'name');
+
+  // Create new counters where applicable
+  for (const name of names) {
+    if (!existingCounters.includes(name)) {
+      await connection<ViewCounter>('viewCounters')
+        .insert({
+          name,
+          count: 1
+        });
+    }
+  }
+
+  return await getCounters(connection, names);
+});
+
+export const decrementCounters = tryDb<[string[]], Record<string, number>>(async (connection, names) => {
+  // Update existing counters
+  const existingCounters = await connection<ViewCounter>('viewCounters')
+    .whereIn('name', names)
+    .decrement('count')
+    .update({}, 'name');
+
+  // Create new counters where applicable
+  for (const name of names) {
+    if (!existingCounters.includes(name)) {
+      await connection<ViewCounter>('viewCounters')
+        .insert({
+          name,
+          count: 0
+        });
+    }
+  }
```

Comment:
Doesn't look like this is used?

Reply:
No, but the ticket specified it, so I assumed it was for future use.  Should I remove?

Reply:
Yes please, the intention is just to increment counters